### PR TITLE
[Doc] [runtime env] Clarify zip requirements for remote URIs

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?

The requirements that a remote URI have a top-level directory are specified fully in the documentation; this PR makes the instructions more foolproof. This PR 
- specifies that the `zip` command must be run from the parent directory
- gives an example of the correct `zipinfo` output
- Adds instructions to the error message when there's no top-level directory

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-core/handling-dependencies.rst
Comment (shrekris-anyscale) [inline doc/source/ray-core/handling-dependencies.rst:513]:
```suggestion
  You can avoid this by using the ``zip -r`` command directly on the directory you want to compress. Make sure to run the command from that directory's parent.
```

**Review by shrekris-anyscale — APPROVED** [review]
Nice update!

File: doc/source/ray-core/handling-dependencies.rst
Comment (architkulkarni) [reply doc/source/ray-core/handling-dependencies.rst:513]: Thanks, this is better

**Comment by architkulkarni** [comment]
Tune test_cluster flaky on master and unrelated

Linkcheck failure ` Forbidden for url: https://www.kdnuggets.com/2019/11/speed-up-pandas-4x.html` unrelated

"C++ and Java" flaky on master

**Comment by edoakes** [comment]
Need codeowner approval

**Comment by architkulkarni** [comment]
@ray-project/ray-docs Can you please approve this as codeowner?

**Review by pcmoritz — APPROVED** [review]

**Comment by pcmoritz** [comment]
Approved, thanks for updating the error messages and documentation, this is great :)

**Pull request merged**
